kvmagent/kvmagent/plugins/nvram/nvram.py

```
@@ -0,0 +1,86 @@
+            if self.nvram_install_path == '':
+                raise Exception('vm_uuid and nvram_install_path in NvRamVmExtensions cannot both be empty.')
+            if not nvram_common.is_nvram_install_path(self.nvram_install_path):
+                return
+            
+            if nvram_sblk.is_sharedblock_install_path(self.nvram_install_path):
+                self.vm_uuid = nvram_sblk.find_vm_uuid_by_sharedblock_install_path(self.nvram_install_path)
+            else: # expect local storage
+                self.vm_uuid = nvram_local.find_vm_uuid_by_nvram_install_path(self.nvram_install_path)
+
+        if not self.vm_uuid:
+            raise Exception("vm_uuid is not set")
```

Ruff (0.15.2)

[warning] Create your own exception

(TRY002)